server/src/calendar/continous_scraping.rs

```diff
@@ -0,0 +1,287 @@
+        let results: Vec<Result<usize, usize>> = join_all(futures).await;
+        results
+            .iter()
+            .map(|e| match e {
+                Ok(cnt) => *cnt,
+                Err(cnt) => *cnt,
+            })
+            .for_each(|e| entry_stats.push(e as u32));
+        // if one of the futures needed to be retried smaller, this would skew the stats a lot
+        if results.iter().all(|e| e.is_ok()) {
+            time_stats.push(round_start_time.elapsed());
+        }
+        if i % 30 == 0 {
+            info!(
+                "Scraped {:.2}% (avg {:.1?}/key, total {:.1?}) result-{:?} in time-{:.1?}",
+                i as f32 / entry_cnt as f32 * 100.0,
+                start_time.elapsed() / i,
+                start_time.elapsed(),
+                entry_stats,
+                time_stats,
+            );
+        }
+        // sleep to not overload TUMonline.
+        // It is critical for successfully scraping that we are not blocked.
+        sleep(Duration::from_millis(100)).await;
+    }
+
+    info!(
+        "Finished scraping calendar entrys. ({} entries in {}s)",
+        entry_cnt,
+        start_time.elapsed().as_secs_f32()
+    );
+}
+
+fn delete_scraped_results() {
+    let conn = &mut utils::establish_connection();
+    use crate::schema::calendar_scrape::dsl::calendar_scrape;
+    diesel::delete(calendar_scrape)
+        .execute(conn)
+        .expect("Failed to delete calendar");
+}
+
+fn promote_scraped_results_to_prod() {
+    let start_time = Instant::now();
+    let conn = &mut utils::establish_connection();
+    use crate::schema::calendar::dsl::calendar;
+    use crate::schema::calendar_scrape::dsl::{calendar_scrape, status};
+    diesel::delete(calendar)
+        .execute(conn)
+        .expect("Failed to delete calendar");
+    diesel::insert_into(calendar)
+        .values(
+            calendar_scrape
+                .filter(status.eq("fix"))
+                .or_filter(status.eq("geplant")),
+        )
+        .execute(conn)
+        .expect("Failed to insert newly scraped values into db");
+
+    info!(
+        "Finished switching scraping results - prod. ({}s)",
+        start_time.elapsed().as_secs_f32()
+    );
+}
+
+fn get_all_ids() -> Vec<(String, i32)> {
+    let conn = &mut utils::establish_connection();
+
+    use crate::schema::de::dsl::*;
+    // order is just here, to make debugging more reproducible. Performance impact is engligable
+    let res = de
+        .select((key, tumonline_room_nr))
+        .filter(tumonline_room_nr.is_not_null())
+        .order_by((key, tumonline_room_nr))
+        .load::<(String, Option<i32>)>(conn);
+    match res {
+        Ok(d) => d
+            .iter()
+            .map(|(k, t)| (k.clone(), t.unwrap()))
+            .collect::<Vec<(String, i32)>>(),
+        Err(e) => {
+            error!("Error requesting all ids: {:?}", e);
+            vec![]
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct ScrapeOrder {
```

Comment:
I'm not sure if I understand the name of this class. Do you mean "order" like an order in a restaurant or like the order of a sequence. Maybe `ScrapeTask` is better

Reply:
True. Probably the explicit `ScrapeRoomTask` is even better.